### PR TITLE
ROC-5473 - Events, Case Event, Complex Types for Claim Submission

### PR DESCRIPTION
**Pull request by karthikkannankainos**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/ROC-5473

### Change description ###
Claim Submission Async operation changes

### CODE REVIEW

File: definition/data/sheets/AuthorisationCaseField.json [hunk 1]
```diff
@@ -165,5 +166,6 @@
   {"CaseTypeID": "MoneyClaimCase", "CaseFieldID": "SOTLabel", "UserRole": "letter-holder", "CRUD": "CRUD"},
   {"CaseTypeID": "MoneyClaimCase", "CaseFieldID": "submitterLabel", "UserRole": "letter-holder", "CRUD": "CRUD"},
   {"CaseTypeID": "MoneyClaimCase", "CaseFieldID": "caseDocuments", "UserRole": "letter-holder", "CRUD": "CRUD"},
-  {"CaseTypeID": "MoneyClaimCase", "CaseFieldID": "migratedFromClaimStore", "UserRole": "letter-holder", "CRUD": "CRUD"}
+  {"CaseTypeID": "MoneyClaimCase", "CaseFieldID": "migratedFromClaimStore", "UserRole": "letter-holder", "CRUD": "CRUD"},
+  {"CaseTypeID": "MoneyClaimCase", "CaseFieldID": "claimSubmissionOperationIndicators", "UserRole": "letter-holder", "CRUD": "CRUD"}
```
Comment (sabahirfan) [inline definition/data/sheets/AuthorisationCaseField.json]: Don't we need to add entry of `claimSubmissionOperationIndicators ` for `caseworker-cmc` ?

Comment (karthikkannankainos) [reply definition/data/sheets/AuthorisationCaseField.json]: caseworker-cmc , will he create a case in CCD?

File: definition/data/sheets/AuthorisationCaseEvent.json [hunk 2]
```diff
@@ -65,6 +66,9 @@
   {"CaseTypeID": "MoneyClaimCase", "CaseEventID": "LinkSealedClaimDocument", "UserRole": "citizen", "CRUD": "CR"},
   {"CaseTypeID": "MoneyClaimCase", "CaseEventID": "LinkDefendant", "UserRole": "citizen", "CRUD": "CR"},
   {"CaseTypeID": "MoneyClaimCase", "CaseEventID": "TestSupportUpdate", "UserRole": "citizen", "CRUD": "CR"},
+  {"CaseTypeID": "MoneyClaimCase", "CaseEventID": "SendingClaimantNotification", "UserRole": "citizen", "CRUD": "CR"},
+  {"CaseTypeID": "MoneyClaimCase", "CaseEventID": "SendingRPA", "UserRole": "citizen", "CRUD": "CR"},
+  {"CaseTypeID": "MoneyClaimCase", "CaseEventID": "PinGenerationOperations", "UserRole": "citizen", "CRUD": "CR"},
```
Comment (sabahirfan) [inline definition/data/sheets/AuthorisationCaseEvent.json]: I think we will need above three event to be configured for `caseworker-cmc` role as well

Comment (karthikkannankainos) [reply definition/data/sheets/AuthorisationCaseEvent.json]: we are sending authorisation of citizen for sending these events. So dont think we need caseworker-cmc. correct me if I am wrong

File: definition/data/sheets/AuthorisationCaseField.json (hunk 1 above)
Comment (sabahirfan) [reply definition/data/sheets/AuthorisationCaseField.json]: yes, at some point using CCD UI after inversion of control. Currently are maintaining the definition inline with that for all other fields.

Comment (karthikkannankainos) [reply definition/data/sheets/AuthorisationCaseField.json]: have added now for consistency. Done

Comment (sabahirfan) [reply definition/data/sheets/AuthorisationCaseField.json]: Thanks 👍

**Pull request merged**